pystac/__init__.py

```diff
@@ -3,59 +3,72 @@
-from pystac.stac_io import STAC_IO
-from pystac.extensions import Extensions
-from pystac.stac_object import (STACObject, STACObjectType)
-from pystac.media_type import MediaType
-from pystac.link import (Link, HIERARCHICAL_LINKS)
-from pystac.catalog import (Catalog, CatalogType)
-from pystac.collection import (Collection, Extent, SpatialExtent, TemporalExtent, Provider)
-from pystac.item import (Item, Asset, CommonMetadata)
-
-from pystac.serialization import stac_object_from_dict
+from pystac.errors import (
+    STACError,  # type:ignore
+    STACTypeError,  # type:ignore
+    RequiredPropertyMissing,  # type:ignore
+)
+
+from typing import Any, Dict, Optional
+from pystac.version import (
+    __version__,
+    get_stac_version,  # type:ignore
+    set_stac_version,  # type:ignore
+)
+from pystac.stac_io import StacIO  # type:ignore
+from pystac.stac_object import STACObject, STACObjectType  # type:ignore
+from pystac.media_type import MediaType  # type:ignore
+from pystac.link import Link, HIERARCHICAL_LINKS  # type:ignore
+from pystac.catalog import Catalog, CatalogType  # type:ignore
+from pystac.collection import (
+    Collection,  # type:ignore
+    Extent,  # type:ignore
+    SpatialExtent,  # type:ignore
+    TemporalExtent,  # type:ignore
+    Provider,  # type:ignore
+    Summaries,  # type:ignore
+    RangeSummary,  # type:ignore
+)
+from pystac.item import Item, Asset, CommonMetadata  # type:ignore
 
 import pystac.validation
+from pystac.validation import STACValidationError  # type:ignore
 
-STAC_IO.stac_object_from_dict = stac_object_from_dict
-
-from pystac import extensions
+import pystac.extensions.hooks
+import pystac.extensions.datacube
 import pystac.extensions.eo
+import pystac.extensions.file
+import pystac.extensions.item_assets
 import pystac.extensions.label
 import pystac.extensions.pointcloud
 import pystac.extensions.projection
 import pystac.extensions.sar
 import pystac.extensions.sat
 import pystac.extensions.scientific
-import pystac.extensions.single_file_stac
 import pystac.extensions.timestamps
 import pystac.extensions.version
 import pystac.extensions.view
-import pystac.extensions.file
-
-STAC_EXTENSIONS = extensions.base.RegisteredSTACExtensions([
-    extensions.eo.EO_EXTENSION_DEFINITION, extensions.label.LABEL_EXTENSION_DEFINITION,
-    extensions.pointcloud.POINTCLOUD_EXTENSION_DEFINITION,
-    extensions.projection.PROJECTION_EXTENSION_DEFINITION, extensions.sar.SAR_EXTENSION_DEFINITION,
-    extensions.sat.SAT_EXTENSION_DEFINITION, extensions.scientific.SCIENTIFIC_EXTENSION_DEFINITION,
-    extensions.single_file_stac.SFS_EXTENSION_DEFINITION,
-    extensions.timestamps.TIMESTAMPS_EXTENSION_DEFINITION,
-    extensions.version.VERSION_EXTENSION_DEFINITION, extensions.view.VIEW_EXTENSION_DEFINITION,
-    extensions.file.FILE_EXTENSION_DEFINITION
-])
-
 
-def read_file(href):
+EXTENSION_HOOKS = pystac.extensions.hooks.RegisteredExtensionHooks(
```

Comment:
I'd guess that is would make more sense to have `EXTENSION_HOOKS` set in pystac/extensions/__init__.py and just do `from extensions import EXTENSION_HOOKS` (or what ever is the most accepted way to refer to files in the same package)

Reply:
Tried this, doing it a couple different ways led to a lot of circular import headaches, so will keep as is